core/src/main/java/org/kohsuke/stapler/lang/KlassNavigator.java

```diff
@@ -78,15 +80,28 @@
 
     /**
      * List fields of this class.
-     *
      * This list excludes fields from super classes.
+     * @param clazz Class
+     * @return List of the fields declared for the class.
+     *         By default this list is empty, {@link KlassNavigator} implementations are responsible to implement it.
+     * @since 1.246
      */
-    public abstract List<FieldRef> getDeclaredFields(C clazz);
+    @Nonnull
+    public List<FieldRef> getDeclaredFields(C clazz) {
+        return Collections.emptyList();
+    }
 
     /**
      * Reports all the methods that can be used for routing requests on this class.
+     * @param clazz Class
+     * @return List of the fields functions declared for the class.
+     *         By default this list is empty, {@link KlassNavigator} implementations are responsible to implement it.
+     * @since 1.246
```

Comment:
BTW `getDeclaredMethods` was also added as `abstract` somewhat recently (1.220), but I suppose if this caused problems we would have hit them long since.